reactExercise/solutions/zihe/ex8/src/NameInfo.js

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import Spinner from './spinner';
+import { Chart as ChartJS, ArcElement, Tooltip, Legend } from 'chart.js';
+import { Pie } from 'react-chartjs-2';
+ChartJS.register(ArcElement, Tooltip, Legend);
+
+function NameInfo() {
+    const [name, setName] = useState('');
+    const [age, setAge] = useState(null);
+    const [gender, setGender] = useState(null);
+    const [nation, setNation] = useState([[], [], [], [], []]);
```

Comment:
这里你相当于在默认这个数据会是 5个array? 这样做不太好 因为API不一定是一直会return 5个结果的.   假设你输入一个很奇怪的名字 asddsad   你的网页就会报错. 原因就是数据不够 但是下面
```
[nationalizeData.country[4].country_id, nationalizeData.country[4].probability]
```
 会去尝试读取数据 然后报 undefined error

Reply:
要把代码写的更灵活一点 
比如说 你让initial value = []
之后读取的时候检测length  如果没有 就不要读取,  如果有 就通过loop index 读取

instead of 
```
nation[0][0]
```
你可以先检测  
```
nation.length > 0 && nation[0].country_id    // 假设你储存的是object
```